modules/ROOT/pages/customer-connector-upgrade.adoc

```diff
@@ -0,0 +1,162 @@
+
+* *Minimum Mule version caveats*
++
+If your custom connector is running on Mule 4.5.x or earlier, you can’t run Java 17 tests that extend the MuleArtifactFunctionalTestCase, but you can still run Java 8 tests.
++
+To decouple the minimum Mule version when running tests, add the `-Dmunit.jvm` parameter to use Java 17 as described in <<test-your-custom-connector-with-munit>>.
+
+[[upgrade-your-mule-sdk-connectors]]
+== Upgrade Your Custom Mule SDK Connectors
+
+Upgrade your custom Mule SDK (Java SDK and XML SDK) connectors to Java 17.
+
+=== Upgrade Third-Party Libraries
+
+Upgrade your custom connector’s third-party libraries to be compatible with Java 17.
+
+* Java EE libraries
++
+Mule 4.6 exposes Java EE libraries differently. You must add the following BOM dependency to your custom connector and, if applicable, ensure the provided conflicting library is properly excluded. You can also add the libraries in the BOM dependency separately.
++
+[source,java,linenums]
+----
+<muleJavaEeBomVersion>4.6.0-20231025</muleJavaEeBomVersion>
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>org.mule</groupId>
+        <artifactId>mule-javaee-runtime-bom</artifactId>
+        <version>${muleJavaEeBomVersion}</version>
+        <type>pom</type>
+        <scope>import</scope>
+      </dependency>
+    </dependencies>
+  </dependencyManagement>
+----
+
+* External libraries
++
+Upgrade external libraries, which are third-party libraries that are not bundled with your custom connector. Some examples include a JDBC driver, JMS broker client, or Groovy runtime. The following example shows a JDBC driver external library configuration: 
++
+[source,java,linenums]
+----
+@ExternalLib(name = "MySQL JDBC Driver",
+description = "A JDBC driver that supports connection to the MySQL Database",
+nameRegexpMatcher = "(.*)\\.jar",
+requiredClassName = "com.mysql.jdbc.Driver",
+coordinates = "mysql:mysql-connector-java:5.1.44")
+public class MySqlConnectionProvider implements ConnectionProvider<Connection> {
+  //
+}
+----
++
+You must update your external libraries to a version that is compatible with Java 17. 
+
+=== Review Your Code
+
+Even though you compile your custom connector with Java 8, ensure you can run your custom connector with Java 17. Reviewing your code is important because there are some restrictions that are flagged as warnings in Java 11 but are flagged as failures and can’t be disabled in Java 17. For example, reflection is more restricted in Java 17, so you must review your code to ensure there are no illegal actions. 
+
+Learn about the different changes in Java 17 to review your code effectively:
+
+* JDK Migration Guide
++
+See the https://docs.oracle.com/en/java/javase/17/migrate/getting-started.html#GUID-C25E2B1D-6C24-4403-8540-CFEA875B994A[Oracle JDK Migration Guide]. It contains for a full list of the changes made in Java 17. Use the guide to assess the impact to your code.
+
+* Mule Runtime Changes
```

Comment:
"Mule Runtime" should probably be "Mule runtime engine". Check the Term Ref:
https://confluence.internal.salesforce.com/display/MTDT/Term+Presentation+Reference